### PR TITLE
`stake-tracker` remote externalities tests + improvements

### PR DESCRIPTION
**Pull request by gpestana**
Changes in the migration code introduced in this PR:
- Migration removes duplicate nominations in all nominators, if they exist (changes by calling `fn do_add_nominator` with dedup nominations)
- Migration removes all the non active validator nominations to avoid adding dangling nominations (changes by calling `fn do_add_nominator` if necessary)
- Migration iterates through all `Nominators` map first and through all the `Validators` map to insert all the validators that are not nominated to the target list (with self-stake only).
- Moves stake-tracker related try-state checks to the staking pallet (no try-state checks in the stake-tracker pallet anymore).
- Runs benchmarks for MMB migration step with bench bot. 

The migration code has been validated against the Polkadot using the externalities tests in [polkadot/runtime/westend/src/lib.rs](https://github.com/paritytech/polkadot-sdk/pull/4436/files/46a80f14ba8990b014d8f9df36727c77e2f3e929#diff-5264d812f7aadfc486efffd3322b11c8dee5e8ddf2e77e0a9e70b573f3d2afdd). Upon running the migrations, we ensure that:
- All validators have been added to the target list with their correct approvals score (as per the try-state checks).
- All nominations are "cleaned" (see def. of clean above)
- Try-state checks related to stake-tracker and approvals pass.

### CODE REVIEW

**Comment by gpestana** [comment]
bot bench polkadot-pallet --runtime=westend --pallet=pallet-staking

**Comment by gpestana** [comment]
bot bench polkadot-pallet --runtime=westend --pallet=pallet_staking

**Comment by kianenigma** [comment]
what is the example execution output? any insights or results?

**Comment by gpestana** [comment]
> what is the example execution output? any insights or results?

The migration and try-state in this branch are working now.

There were a few edge side-cases that we had to change in this PR, I'll add them to the comments and then merge this PR into the main PR branch.

**Comment by gpestana** [comment]
bot bench polkadot-pallet --runtime=westend --pallet=pallet_staking

File: polkadot/runtime/westend/src/lib.rs [hunk 1]
```diff
@@ -2682,6 +2681,64 @@ mod remote_tests {
 			.unwrap();
 		ext.execute_with(|| Runtime::on_runtime_upgrade(UpgradeCheckSelect::PreAndPost));
 	}
+
+	#[tokio::test]
```
Comment (gpestana) [inline polkadot/runtime/westend/src/lib.rs:2685]: TODO: remove these changes before merging in the base branch.

**Comment by gpestana** [comment]
bot bench polkadot-pallet --runtime=westend --pallet=pallet_staking

**Comment by gpestana** [comment]
bot bench polkadot-pallet --runtime=westend --pallet=pallet_staking

**Comment by gpestana** [comment]
bot bench polkadot-pallet --runtime=westend --pallet=pallet_staking

**Comment by gpestana** [comment]
bot cancel 2-1147b240-298f-42c4-93de-6c5f67469675

**Comment by gpestana** [comment]
Closed and non-merged due to history mess up. This PR has been open and merged instead of this, same logic https://github.com/paritytech/polkadot-sdk/pull/4673

**Pull request closed without merging**